flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocator.java

```diff
@@ -94,6 +96,34 @@ class SlotSharingExecutionSlotAllocator implements ExecutionSlotAllocator {
         this.sharedSlots = new IdentityHashMap<>();
     }
 
+    @Override
+    public List<ExecutionSlotAssignment> allocateSlotsFor(
+            List<ExecutionAttemptID> executionAttemptIds) {
+
+        final Map<ExecutionVertexID, ExecutionAttemptID> vertexIdToExecutionId = new HashMap<>();
+        executionAttemptIds.forEach(
+                executionId ->
+                        vertexIdToExecutionId.put(executionId.getExecutionVertexId(), executionId));
+
+        checkState(
+                vertexIdToExecutionId.size() == executionAttemptIds.size(),
+                "SlotSharingExecutionSlotAllocator does not support one execution vertex to have multiple concurrent executions");
+
+        final List<ExecutionVertexID> vertexIds =
```

Comment:
`vertexIds = new ArrayList<>(vertexIdToExecutionId.keySet())`

Reply:
We cannot do this because we need to respect the given order.

Reply:
OK